src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -973,6 +982,26 @@ public void run() {
         }
     }
 
+    /**
+     * Verifies the status messages in the page are same as the messages specified.
+     * The check is done multiple times with waiting times in between to account for
+     * timing issues due to page load, inconsistencies in Selenium API, etc.
+     * @param expectedStatuses : one or more expected status messages passed as varargs.
+     */
+    public void verifyStatus(final String... expectedStatuses) {
```

Comment:
- What happens if I pass in no arguments?
- `verifyStatus` -> `verifyEqualsTextsForAllStatusMessagesToUser`. I can't seem to think of a better name for this so feel free to come up with a better one.
- `expectedStatuses` -> `expectedTexts`

Reply:
>What happens if I pass in no arguments?

Will fix this.

>`verifyStatus` -> `waitForTextsForAllStatusMessagesToUserEquals`. If you have a better name, feel free to change it.

**If we do decide** to remove `verifyStatus(final String expectedStatus) and getStatus()` and only keep one method, do we still need to rename it? As there may be other uses of the method which are not in the scope of this PR, which will also need to be changed.
And **if we don't** remove the above mentioned method, how about `verifyStatus` -> `verifyTextsForAllStatusMessages`?

Reply:
Because you are modifying the method, so it is apt to rename the method to a more appropriate name.

The reason `verify` is not accurate because it is a term we came up ourselves, which doesn't tell others that we are actually asserting. Furthermore, in this case, we are not just "verifying" but rather also waiting before that. The method name should make sure the meaning is conveyed as much as possible (i.e. self documenting).

Reply:
How about `checkStatusMessages`? TBH I think 
- it would be a little misleading to start the name with `wait`, as it will suggest that the main function of this method is to wait, like in `waitForElementsVisibility` and similar methods, where as the main motive to call this function is to check if the expected status messages are same as the actual ones or not
- `waitForTextsForAllStatusMessagesToUserEquals` is too verbose, developers may find it difficult to use it?

waiting for your opinion on this one.

Reply:
@sukanta-27 It's very nice for you to engage in thought and discussion :).

**So first let's see the problem you are highlighting:**

Is verbosity worse because the developer is forced to read the **complex** semantics by reading the method name?

Or is it better to just let the developer assume the meaning of the method or being forced to read the documentation to find out the meaning?

**Then let's look at your points one by one:**

> How about checkStatusMessages?

What does `check` mean here? Am I checking if the status messages are visible? Am I checking if the status messages are present? Am I checking if the status messages contain something? 

Wouldn't this force me to guess the meaning or read the documentation to understand the method?

> it would be a little misleading to start the name with wait

Starting with `wait` does not necessarily imply that the main function is waiting. However, it does imply a certain importance to it. In this case, `wait` is of primary importance because of the time it takes and the side effects it results in (such as throwing time out exceptions). 

Would you say then then we should hide the fact that we need to wait for the status message to become equal or just simply say we are checking that the status messages are equal? Are these two semantically the same?

Now to address the more subtle point of verbosity, this can be avoided if we have better separation of concerns.

For example, take the following simplified scenario:
1. Click submit on page.
2. Wait for page to finish loading.
3. Scroll to status message, we assume it exists.
4. Now we do `verifyTextsForAllStatusMessagesEquals`. (We assume we can get rid of `ToUser` by having better separation of concerns)

If we break down the method:
- It is not clear what `verify` means. I'm sure there is a better word for it but I don't know. `assert` is better when we have two parameters but we have one here.
- `text` is still something one has to learn about HTML but at least it's guessable if one knows the implementation of status messages
- We are coupled to the implementation of status messages. Is there a way around it? I don't know.
- `Equals` is as clear as it gets.

Eventually, projects are going to get more complex and trying to reduce coupling gets harder or sometimes not practical or impossible. In those cases, it's easy to make the decision to shorten the method name for the sake of "clean code", forcing developers to read the documentation to understand the complex method names or worse assume/guess the meanings of methods.

For more discussions: https://softwareengineering.stackexchange.com/questions/169925/are-long-methods-always-bad/169930

You can look for more if you are interested.

Reply:
@LiHaoTan Thank you very much for explaining it to me so elaborately. Cleared my concerns about verbosity.

>What does check mean here? Am I checking if the status messages are visible? Am I checking if the status messages are present? Am I checking if the status messages contain something?

It agree that check will bring a lot of ambiguity to the method name, which will cause confusion. 

>In this case, wait is of primary importance because of the time it takes and the side effects it results in (such as throwing time out exceptions).

overlooked the fact that using this method may have side effects as it is not just asserting but also waiting. which the developer do need to know. 

Keeping these in mind, Can I then instead of breaking down the method, keep the name you suggested earlier? :sweat_smile:

Reply:
> Can I then instead of breaking down the method, keep the name you suggested earlier?

Of course, my idea is to try to minimize changes while keeping the code at reasonable quality. You are always free to improve upon the ideas if you have any.